src/lens.c

```diff
@@ -2695,19 +2695,45 @@ static LVINFO_UPDATE_FUNC(picq_update)
 
     if (!is_movie_mode())
     {
-        // kitor FIXME: Not true on R anymore
-        // Leads to werid values like RAW+L is "RAW+Err", RAW+M is "mRAW"
+#ifdef CONFIG_DIGIC_VIII
+/* via R.180, confirmed RP.160; M50 and 850D have the same set of modes:
+ * L        03030100   .XX .... ..XX .... ...X .... ....
+ * l        03020100   .XX .... ..X. .... ...X .... ....
+ * M        03030101   .XX .... ..XX .... ...X .... ...X
+ * m        03020101   .XX .... ..X. .... ...X .... ...X
+ * S1       0303010E   .XX .... ..XX .... ...X .... XXX.
+ * s1       0302010E   .XX .... ..X. .... ...X .... XXX.
+ * S2       0303010F   .XX .... ..XX .... ...X .... XXXX
+ * CRAW     03030600   .XX .... ..XX .... .XX. .... ....
+ * RAW      04030600   X.. .... ..XX .... .XX. .... ....
+ * Some combinations:
+ *  RAW + L 04030700   X.. .... ..XX .... .XXX .... ....
+ *  RAW + l 04020700   X.. .... ..X. .... .XXX .... ....
+ * CRAW + L 03030700   .XX .... ..XX .... .XXX .... ....
+ * CRAW + l 03020700   .XX .... ..X. .... .XXX .... ....
+ */
+        int raw = pic_quality & 0x600; // 3 for *RAW, 0 - not RAW
+        int jpg = (pic_quality & 0x100);
+        int rawsize = pic_quality >> 26; // if raw: 0 CRAW, 1 RAW
+        int jpegtype = (pic_quality >> 16) & 0xF;
+        int jpegsize = pic_quality & 0xFF;
+#else
```

Comment:
In fact everything except `rawsize` seems to be shifted by 8. `rawsize` is probably moved, but with just two options and MSBs not used for anything it can't be assumed that it is still 8 bits.

Conditionally shifting would still require ifdef for that shift + ifdefs for `rawsize`, I'm not sure if this is worth it?

Reply:
I'm not sure either, use whichever you think is cleaner.

Reply:
Then I'll stay as is. Old way has no documentation / proof dumped from real camera - we just have to trust the code it is ok. New has all the examples put in comment, verified R/RP/M50. If old one would be documented well, I'd reconsider - but just for that I'd keep them separated.